enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/OnboardingProcessRepository.java

```diff
@@ -36,10 +36,8 @@
 @Repository
 public interface OnboardingProcessRepository extends CrudRepository<OnboardingProcessEntity, String> {
 
-    @Query("SELECT p FROM OnboardingProcessEntity p WHERE p.status = :status " +
-            "AND p.userId = :userId " +
-            "ORDER BY p.timestampCreated DESC")
-    Optional<OnboardingProcessEntity> findExistingProcessForUser(String userId, OnboardingStatus status);
+    @Query("SELECT p FROM OnboardingProcessEntity p WHERE p.status = :status AND p.identificationData = :identificationData")
```

Comment:
This code may not work unless we normalize the way JSON data is serialized in the database. We should consider:

```java
ObjectMapper mapper = new ObjectMapper();
mapper.setConfig(mapper.getSerializationConfig()
   .with(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY));
   .with(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS)
```

Reply:
... plus we should probably ensure that the other typical serialization features (formatting, date serialization, etc.) work in the same manner every time...